Fixes missing materials upon deconstruction/retrieval

TehZombehz opened the pull request:
Fixes attempts to retrieve materials resulting in missing stacks if there is more than one full stack of a material in the machine.
This fix should affect anything that can contain more than one stack of materials (autolathe, exosuit fabricator, and so on).

Fixes  #22506.

TehZombehz committed:
Fixes missing materials upon retrieval

Fixes attempts to retrieve materials resulting in missing stacks if
there is more than one full stack of a material in the machine.

RemieRichards commented on code/datums/material_container.dm:
```diff
@@ -143,13 +143,12 @@
 			use_amount_type(sheet_amt * MINERAL_MATERIAL_AMOUNT, M.id)
 			sheet_amt -= MAX_STACK_SIZE
-
+			M.amount = (sheet_amt * MINERAL_MATERIAL_AMOUNT)
```
should this not be before the line above?

TehZombehz replied:
You mean above line 150?

RemieRichards replied:
That is what I am referring to, yes.

My query arose due to ```use_amount_type()``` using the same calculation you're using for the sheet's new amount, except since ```sheet_amt``` is modified before you repeat the calculation, these two numbers aren't actually the same (when it *reads* as if they should be)

Otherwise the amount of material in the sheet is less than the amount used, which is weird.

TehZombehz replied:
Does 
>use_amount_type(sheet_amt * MINERAL_MATERIAL_AMOUNT, M.id)

modify M.amount the same way that
>M.amount = (sheet_amt * MINERAL_MATERIAL_AMOUNT) 

does?

TehZombehz committed:
Alternative Fix

This is what actually needed fixing. M.amount would always equal 0, so
round would always result in 0, thus sheet remainders would never get
outputted. This resolves that.
